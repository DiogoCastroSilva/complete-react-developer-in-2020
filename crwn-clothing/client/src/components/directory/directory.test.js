import React from 'react';
import { shallow } from 'enzyme';

import Directory from './directory.component';
import MenuItem from '../menu-item/menu-item.component';

describe('testing Directory component', () => {
    let wrapper;

    beforeEach(() => {
        const mockProps = {
            sections: [
                {
                    title: 'Page',
                    imageUrl: 'https://i.ibb.co/ZYW3VTp/brown-brim.png',
                    linkUrl: 'Page'
                }
            ]
        };

        wrapper = shallow(<Directory {...mockProps} />);
    });

    it('should render a Directory component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have one MenuItem component', () => {
        expect(wrapper.find(MenuItem).length).toEqual(1);
    });

    it('should not have any MenuItem component', () => {
        const withItemsMockProps = {
            sections: []
        };

        const newWrapper = shallow(<Directory {...withItemsMockProps} />);

        expect(newWrapper.find(MenuItem).exists()).toEqual(false);
    });
});