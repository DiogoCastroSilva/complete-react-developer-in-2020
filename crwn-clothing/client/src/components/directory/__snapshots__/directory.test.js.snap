// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Directory component should render a Directory component 1`] = `
<styled.div>
  <withRouter(MenuItem)
    imageUrl="https://i.ibb.co/ZYW3VTp/brown-brim.png"
    linkUrl="Page"
    title="Page"
  />
</styled.div>
`;
